worker/workertest/superseded_test.go

```diff
@@ -4,19 +4,51 @@ import (
 	"encoding/json"
 	"net/http"
 	"net/http/httptest"
+	"sync"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/taskcluster/slugid-go/slugid"
+	"github.com/taskcluster/taskcluster-client-go/queue"
 	"github.com/taskcluster/taskcluster-worker/runtime"
 )
 
 func TestSuperseded(t *testing.T) {
+	// This test aims to check if the superseded functionality works. As we can't
+	// know which order the tasks will be claimed in... This a little complicated
+	// to test. We do it by using the task index to recall which one of them set
+	// to be superseded. When the HTTP server gets called as supersederUrl and
+	// we haven't returned a superseded task yet, we take the next available taskId
+	// as superseded the current taskId. We store the index of the superseded task
+	// as superseded.
+	superseded := -1 // index of task that was superseded
 	var c Case
+	var m sync.Mutex
 	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		m.Lock() // lock access to superseded
+		defer m.Unlock()
+
+		// Get taskId from which this is being called
 		taskID := r.URL.Query().Get("taskId")
+
+		// Create array of superseding taskIds, always start with it's own taskId
 		supersedes := []string{taskID}
-		if c.Tasks[0].TaskID == taskID {
-			supersedes = append(supersedes, c.Tasks[1].TaskID)
+
+		// If we haven't made a superseded task yet we do that now
+		if superseded == -1 {
+			// Find index of this taskId
+			for index, task := range c.Tasks {
+				if task.TaskID == taskID {
+					superseded = index
```

Comment:
shouldn't we add a `break` statement here?

Reply:
No harm either way.